### PR TITLE
Update promise method descriptions

### LINKED ISSUE #27515: The description about `Promise.any()` is a bit confusing
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise

### What specific section or headline is this issue about?

[`Promise.any()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise#promise.any_2)

### What information was incorrect, unhelpful, or incomplete?

Promise methods like `Promise.any()`, `Promise.race()`, etc, would actually return a new promise immediately. But the description about `Promise.any()` is:

> Takes an iterable of Promise objects and, as soon as one of the promises in the iterable fulfills, returns a single promise that fulfills with the value from that promise.

this may confuse readers: whether this method return a promise after one of the promises fulfilled (or return a promise immediately and fulfills when one of the promises is fulfilled)?

### What did you expect to see?

clearer description, the promise would be returned immediately.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

reference: https://github.com/mdn/translated-content/pull/13809#discussion_r1236993877

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/promise`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/promise/index.md
* Last commit: https://github.com/mdn/content/commit/01bf58c144237c1fc75cf7135491346778591f09
* Document last modified: 2023-04-08T17:33:52.000Z

</details>

### PR DESCRIPTION
**Pull request by Josh-Cena**
### Description

Fix https://github.com/mdn/content/issues/27515

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/promise/index.md [hunk 1]
```diff
@@ -181,29 +181,13 @@ Note that JavaScript is [single-threaded](/en-US/docs/Glossary/Thread) by nature
...
 
 - {{jsxref("Promise.all()")}}
-
-  - : Wait for all promises to be fulfilled, or for any to be rejected.
-
-    If the returned promise fulfills, it is fulfilled with an aggregating array of the values from the fulfilled promises, in the same order as defined in the iterable of multiple promises.
-
-    If it rejects, it is rejected with the reason from the first promise in the iterable that was rejected.
-
+  - : Takes an iterable of promises as input and returns a single {{jsxref("Promise")}}. This returned promise fulfills when all of the input's promises fulfill (including when an empty iterable is passed), with an array of the fulfillment values. It rejects when any of the input's promises rejects, with this first rejection reason.
```
Comment (hamishwillee) [inline files/en-us/web/javascript/reference/global_objects/promise/index.md:184]:
```suggestion
  - : Takes an iterable of promises as input and returns a single {{jsxref("Promise")}}. This returned promise fulfills when all of the input's promises fulfill (including when an empty iterable is passed), with an array of the fulfillment values. It rejects when any of the input's promises reject, with this first rejection reason.
```

File: files/en-us/web/javascript/reference/global_objects/promise/index.md [hunk 2]
```diff
@@ -181,29 +181,13 @@ Note that JavaScript is [single-threaded](/en-US/docs/Glossary/Thread) by nature
...
 - {{jsxref("Promise.allSettled()")}}
-
-  - : Wait until all promises have settled (each may fulfill or reject).
-
-    Returns a Promise that fulfills after all of the given promises is either fulfilled or rejected, with an array of objects that each describe the outcome of each promise.
-
+  - : Takes an iterable of promises as input and returns a single {{jsxref("Promise")}}. This returned promise fulfills when all of the input's promises settle (including when an empty iterable is passed), with an array of objects that describe the outcome of each promise.
 - {{jsxref("Promise.any()")}}
-  - : Takes an iterable of Promise objects and, as soon as one of the promises in the iterable fulfills, returns a single promise that fulfills with the value from that promise.
+  - : Takes an iterable of promises as input and returns a single {{jsxref("Promise")}}. This returned promise fulfills when any of the input's promises fulfills, with this first fulfillment value. It rejects when all of the input's promises reject (including when an empty iterable is passed), with an {{jsxref("AggregateError")}} containing an array of rejection reasons.
```
Comment (hamishwillee) [inline files/en-us/web/javascript/reference/global_objects/promise/index.md:188]:
```suggestion
  - : Takes an iterable of promises as input and returns a single {{jsxref("Promise")}}. This returned promise fulfills when any of the input's promises fulfill, with this first fulfillment value. It rejects when all of the input's promises reject (including when an empty iterable is passed), with an {{jsxref("AggregateError")}} containing an array of rejection reasons.
```

**Comment by Josh-Cena** [comment]
@hamishwillee, "any of X" is a singular, so we should use singular here.

> Does any of your classmates drink?

File: files/en-us/web/javascript/reference/global_objects/promise/index.md [hunk 3]
```diff
@@ -181,29 +181,13 @@ Note that JavaScript is [single-threaded](/en-US/docs/Glossary/Thread) by nature
...
 - {{jsxref("Promise.allSettled()")}}
-
-  - : Wait until all promises have settled (each may fulfill or reject).
-
-    Returns a Promise that fulfills after all of the given promises is either fulfilled or rejected, with an array of objects that each describe the outcome of each promise.
-
+  - : Takes an iterable of promises as input and returns a single {{jsxref("Promise")}}. This returned promise fulfills when all of the input's promises settle (including when an empty iterable is passed), with an array of objects that describe the outcome of each promise.
 - {{jsxref("Promise.any()")}}
-  - : Takes an iterable of Promise objects and, as soon as one of the promises in the iterable fulfills, returns a single promise that fulfills with the value from that promise.
+  - : Takes an iterable of promises as input and returns a single {{jsxref("Promise")}}. This returned promise fulfills when any of the input's promises fulfill, with this first fulfillment value. It rejects when all of the input's promises reject (including when an empty iterable is passed), with an {{jsxref("AggregateError")}} containing an array of rejection reasons.
```
Comment (hamishwillee) [inline files/en-us/web/javascript/reference/global_objects/promise/index.md:188]: So for the `any()` case, unlike the others, passing an empty iterable causes rejection?

I'll set as "approved" on the assumption that what is written is correct.

**Review by hamishwillee — APPROVED** [review]
This is very clear.

I assume the promise-method-specific files do not have the same confusion as in this top level file?

**Comment by Josh-Cena** [comment]
Right, I simply copied the intro paragraphs there to here ^^

**Comment by hamishwillee** [comment]
> @hamishwillee, "any of X" is a singular, so we should use singular here.

Yes. That is the rule. Though in truth, I didn't think about it - just know that it "sounds more correct". I come from a generation of school kids that weren't taught grammar so I do get it wrong sometimes.

> Does any of your classmates drink?

:-) 

If you're drinking, keep it up. This is good.

**Pull request merged**